src/Factory.php

```diff
@@ -202,6 +205,13 @@ final public function instantiateWith(callable $instantiator): self
         return $cloned;
     }
 
+    final public function setCascadePersist(bool $cascadePersist): self
```

Comment:
I'd prefer this not be part of the public api. Just access and set the private property directly.